File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var postorderTraversal = function(root) {
    const res = [];
    fun(root, res);
    return res;
};

function fun(root, res){
    // Base Case:
    if(root == null) return;
    
    // First, Left Tree:
    fun(root.left, res);
    // Second, right Tree:
    fun(root.right, res);
    // Third, Root
    res.push(root.val);
}

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        helper(root, res);
        return res;
    }
    
    void helper(TreeNode root, List<Integer> res){
        // Base Case:
        if(root == null) return;
        
        // First, Left Sub Tree:
        helper(root.left, res);
        // Second, Right Sub Tree:
        helper(root.right, res);
        // Third, Putting the Root value to the List:
        res.add(root.val);
    }
}